api/src/main/java/marquez/db/ColumnLineageDao.java

```diff
@@ -184,11 +184,7 @@ SELECT DISTINCT ON (cl.output_dataset_field_uuid, cl.input_dataset_field_uuid) c
           WHERE ARRAY[<values>]::DATASET_NAME[] && dv.dataset_symlinks -- array of string pairs is cast onto array of DATASET_NAME types to be checked if it has non-empty intersection with dataset symlinks
           ORDER BY output_dataset_field_uuid, input_dataset_field_uuid, updated_at DESC, updated_at
         ),
-        dataset_fields_view AS (
-          SELECT d.namespace_name as namespace_name, d.name as dataset_name, df.name as field_name, df.type, df.uuid
-          FROM dataset_fields df
-          INNER JOIN datasets_view d ON d.uuid = df.dataset_uuid
-        )
+        dataset_fields_view AS ( SELECT d.namespace_name as namespace_name, d.name as dataset_name, df.name as field_name, df.type, df.uuid FROM dataset_fields df INNER JOIN ( select * from datasets_view where current_version_uuid IN ( SELECT DISTINCT output_dataset_version_uuid FROM selected_column_lineage UNION SELECT DISTINCT input_dataset_version_uuid FROM selected_column_lineage ) ) d ON d.uuid = df.dataset_uuid )
```

Comment:
Can you apply some formatting here like you did in the PR review?

Also when you say `select *`, do we really need to select all the columns of the view?

Reply:
Hi @phixMe It was resolved. Please help to review it again.